connector/dialect.tdd

```diff
@@ -379,61 +138,34 @@
       <formula>CAST(%1 AS TIMESTAMP)</formula>
       <argument type='datetime' />
     </function>
-    <function group='cast' name='FLOAT' return-type='real'>
+    <function group='cast' name='STR' return-type='str'>
```

Comment:
I don't see all of the various casts that we say we support in mongosql (i.e. what about casting a string to an int or a real or a bool?) Do we need to enumerate all possible valid casts?

Reply:
I added the ones I see we support

Reply:
There are still several missing cast functions missing:
- all types to `BOOL`
- all types to `DOUBLE` (kinda discussed in previous comments)
- We also support casting to ARRAY, OBJECT, LONG, and OBJECTID, but I'm not sure how we'd represent those in this file since there are not clean `return-type`s for those.

[This section](https://docs.google.com/document/d/14DnyMSzVNRkoyib8yM62MPJIrkrG1pE0KjRYOWOUuDc/edit#heading=h.vjgnkzhj81tw) of the syntax and semantics doc covers everything!

Reply:
Added the `BOOL`, `DECIMAL`, `LONG`, and additional casts for `DATETIME`, `FLOAT`.  The ones without a matching return type or arguments were not added.